Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -42,6 +42,53 @@ namespace detail
         return make_pair(record_name, component_name);
     }
 
+    /** Return the components of particle positions
+     */
+    inline std::vector< std::string >
+    getParticlePositionComponents()
+    {
+        using vs = std::vector< std::string >;
+#if AMREX_SPACEDIM==3
+        vs const positionComponents{"x", "y", "z"};
+#elif defined(WARPX_DIM_RZ)
+        vs const positionComponents{"r", "z"};
+#else
+        vs const positionComponents{"x", "z"};
+#endif
+        return positionComponents;
+    }
+
+    /** Return the axis (index) names of a mesh
+     */
+    inline std::vector< std::string >
+    getFieldAxisLabels()
+    {
+        using vs = std::vector< std::string >;
+#if AMREX_SPACEDIM==3
+        vs const axisLabels{"x", "y", "z"};
+#elif defined(WARPX_DIM_RZ)
+        vs const axisLabels{"r", "z"};
+#else
+        vs const axisLabels{"x", "z"};
+#endif
+        return axisLabels;
+    }
+
```

Comment:
The two functions above are identical. Could they be replaced by one function with a name like getAxisNames?

Also, for the XZ case, do you want to check #elif defined(WARPX_DIM_XZ)?

Otherwise looks good.

Reply:
No, one is for fields axis names and one for particle component names. This will  change for particles after transformation in the future.

Reply:
> Also, for the XZ case, do you want to check #elif defined(WARPX_DIM_XZ)?

This should be covered by the `#else` but using the `defined(WARPX_DIM_XZ)` would make it more explicit.